src/main/java/uk/gov/companieshouse/service/dissolution/DissolutionEmailService.java

```diff
@@ -90,6 +88,28 @@ public void sendPendingPaymentEmail(Dissolution dissolution) {
         sendEmail(emailDocument);
     }
 
+    public void notifySignatoriesToSign(Dissolution dissolution) {
+        final MessageType messageType = messageTypeCalculator.getForSignatoriesToSign(dissolution);
+
+        final String deadlineDate = deadlineDateCalculator.calculateSignatoryDeadlineDate(dissolution.getCreatedBy().getDateTime());
+
+        getUniqueSignatories(dissolution)
+                .stream()
+                .filter(signatoryEmail -> !signatoryEmail.equals(dissolution.getCreatedBy().getEmail()))
+                .map(signatoryEmail -> mapToSignatoryToSignEmail(dissolution, signatoryEmail, messageType, deadlineDate))
+                .forEach(this::sendEmail);
```

Comment:
can this reuse the function below? Or can both public functions call onto a common private function? A lot of duplication between these two functions

Reply:
How would see it reused? I'm having trouble preventing deadline and message type being recalculated for each email.

Reply:
Hmm good point, maybe only two lines could be reused in both methods i.e.
```

@@ -22,8 +22,6 @@
import java.util.List;
import java.util.stream.Collectors;

import static uk.gov.companieshouse.util.DateTimeGenerator.generateCurrentDateTime;

@Service
public class DissolutionEmailService {

@@ -90,6 +88,28 @@ public void sendPendingPaymentEmail(Dissolution dissolution) {
        sendEmail(emailDocument);
    }

    public void notifySignatoriesToSign(Dissolution dissolution) {
        final MessageType messageType = messageTypeCalculator.getForSignatoriesToSign(dissolution);

        final String deadlineDate = deadlineDateCalculator.calculateSignatoryDeadlineDate(dissolution.getCreatedBy().getDateTime());

        getUniqueSignatories(dissolution)
                .stream()
                .filter(signatoryEmail -> !signatoryEmail.equals(dissolution.getCreatedBy().getEmail()))
                .forEach(signatoryEmail -> handleSignatoryToSignEmail(dissolution, signatoryEmail, messageType, deadlineDate));

...

        final MessageType messageType = messageTypeCalculator.getForSignatoriesToSign(dissolution);

        final String deadlineDate = deadlineDateCalculator.calculateSignatoryDeadlineDate(dissolution.getCreatedBy().getDateTime());

        handleSignatoryToSignEmail(dissolution, email, messageType, deadlineDate);
```